string-methods/challenges-05.test.js

```diff
@@ -0,0 +1,358 @@
+    'De-prickle the gruffalo',
+    'Sprinkle with cinnamon, sugar, flour, and nuts',
+    'Mix until evenly distributed',
+    'Grease a 3-foot x 3-foot casserole dish',
+    'Combine gruffalo compote with water to maintain moisture in the oven',
+    'Fold together remaining ingredients to make the crisp',
+    'Spread the crisp evenly over the gruffalo mixture',
+    'Bake for 12-15 hours',
+  ]
+};
+
+
+const listFoods = (recipe) => {
+  let result = [];
+
+  recipe.ingredients.forEach(ingredient => {
+    let ingredientParsed = ingredient.split(' ').slice(2).join(' ');
+    result.push(ingredientParsed);
+  });
+
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named splitFoods that uses split to produce the same output as Challenge 3.
+
+You may also use other string or array methods.
+------------------------------------------------------------------------------------------------ */
+
+const splitFoods = (recipe) => {
+  let result = [];
+  // Solution code here...
+  recipe.ingredients.forEach(ingredient => {
+    let ingredientParsed = ingredient.split(' ').slice(2).join(' ');
+    result.push(ingredientParsed);
+  });
+
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Use the same recipe from Challenge 3, above.
+
+Write a function named stepAction that takes in the recipe and extracts the action verbs from the steps. Fortunate for you, the action verbs are the first word of each action.
+
+Return a new array containing just the verbs. For example, ['Mix until evenly distributed'] returns ['Mix'].
+------------------------------------------------------------------------------------------------ */
+
+const stepActions = (recipe) => {
+  //Reference:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr
+
+  //Reference: https://www.w3schools.com/jsref/jsref_indexof.asp
+
+  let result = [];
+  // Solution code here...
+  recipe.steps.forEach(step => {
+    let stepParsed = step.substr(0, step.indexOf(' '));
+    result.push(stepParsed);
+  });
+
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named removeEvenValues that, given an array of integers as input, deletes all even values from the array, leaving no 'gaps' behind.
+
+The array should be modified in-place.
+
+For example:
+  const integers = [1, 2, 3, 4, 5, 6];
+  removeEvenValues(integers);
+  console.log(integers) will print [1, 3, 5]
+------------------------------------------------------------------------------------------------ */
+
+const removeEvenValues = (arr) => {
+  // Solution code here...
+  let array = [...arr];
+
+  arr.splice(0, arr.length);
+
+  array.forEach(a => {
+    if (a % 2 !== 0) {
```

Comment:
Since you have prior coding experince you should check out the & operator (different from &&) to determine if a number is even or odd. It's much faster than using modulo.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators